### PR TITLE
SCRUM-3515 LinkML v2.2.0 migration

### PR DESCRIPTION
**Pull request by markquintontulloch**

### CODE REVIEW

File: src/main/cliapp/src/containers/diseaseAnnotationsPage/DiseaseAnnotationsTable.js
Comment (adamgibs) [inline src/main/cliapp/src/containers/diseaseAnnotationsPage/DiseaseAnnotationsTable.js]: This is one of the lines that's causing the error. Looks like a find and replace error. Should check for a JavaScript "object" type.

File: src/main/cliapp/src/containers/diseaseAnnotationsPage/NewAnnotationForm.js
Comment (adamgibs) [inline src/main/cliapp/src/containers/diseaseAnnotationsPage/NewAnnotationForm.js]: This is one of the lines that's causing the error. Looks like a find and replace error. Should check for a JavaScript "object" type.

File: src/main/cliapp/src/containers/diseaseAnnotationsPage/DiseaseAnnotationsTable.js
Comment (markquintontulloch) [reply src/main/cliapp/src/containers/diseaseAnnotationsPage/DiseaseAnnotationsTable.js]: Got a bit find-and-replace happy! Fixed now

File: src/main/cliapp/src/containers/diseaseAnnotationsPage/NewAnnotationForm.js
Comment (markquintontulloch) [reply src/main/cliapp/src/containers/diseaseAnnotationsPage/NewAnnotationForm.js]: Also fixed now

File: src/main/java/org/alliancegenome/curation_api/model/entities/base/AuditedObject.java
Comment (oblodgett) [inline src/main/java/org/alliancegenome/curation_api/model/entities/base/AuditedObject.java:50]: @markquintontulloch so is this going to be the base class for everything? As in all objects will be getting a "id" field which will be the primary key across the whole DB? Just that AuditedObject will be a MappedSuper vs table.

File: src/main/java/org/alliancegenome/curation_api/model/entities/base/SubmittedObject.java
Comment (oblodgett) [inline src/main/java/org/alliancegenome/curation_api/model/entities/base/SubmittedObject.java:30]: So this being a mapped super will mean that the fields of AO, CO, and SO are all down on the child classes.

Comment (markquintontulloch) [reply src/main/java/org/alliancegenome/curation_api/model/entities/base/SubmittedObject.java:30]: Yeah, that’s intended - too slow otherwise

File: src/main/java/org/alliancegenome/curation_api/model/entities/base/AuditedObject.java
Comment (markquintontulloch) [reply src/main/java/org/alliancegenome/curation_api/model/entities/base/AuditedObject.java:50]: Yeah. We need IDs on the biological entities as some don’t have curies or MOD IDs.  Ontology terms don’t really need them, but I think this gives us more flexibility for the future

File: src/main/java/org/alliancegenome/curation_api/model/entities/GenomicEntity.java
Comment (oblodgett) [inline src/main/java/org/alliancegenome/curation_api/model/entities/GenomicEntity.java:33]: So this relationship type should be ManyToMany given that there is a join table. Which does not support the orphanRemoval. So typically what would be done here is another class is created: GenomicEntityCrossReference which represents the entity in the join table. Then this class has a `List<GenomicEntityCrossReference>` with a OneToMany relationship.

**Comment by adamgibs** [comment]
@markquintontulloch As far as I can tell, the UI code looks good now. I'll let Olin go ahead and approve it when he's ready.

File: src/main/java/org/alliancegenome/curation_api/dao/AffectedGenomicModelDAO.java
Comment (oblodgett) [inline src/main/java/org/alliancegenome/curation_api/dao/AffectedGenomicModelDAO.java]: Does a findByParam not work here? AGMDiseaseAnnotationDAO.findByParams()
map.put("subjectBiologicalEntity.id", value);

Comment (oblodgett) [reply src/main/java/org/alliancegenome/curation_api/dao/AffectedGenomicModelDAO.java]: Or are you only wanting id's back? We need to create a method with the same interface that returns ids vs full objects.

File: src/main/java/org/alliancegenome/curation_api/jobs/util/SlackNotifier.java
Comment (oblodgett) [inline src/main/java/org/alliancegenome/curation_api/jobs/util/SlackNotifier.java:25]: I am guessing this is a change that came in from another PR? I think this should be `, defaultValue=""`

Comment (oblodgett) [reply src/main/java/org/alliancegenome/curation_api/jobs/util/SlackNotifier.java:25]: Or just add `NET = dev` in your application.properties file.

File: src/main/java/org/alliancegenome/curation_api/model/entities/AGMDiseaseAnnotation.java
Comment (oblodgett) [inline src/main/java/org/alliancegenome/curation_api/model/entities/AGMDiseaseAnnotation.java:63]: @cmpich this is the code I am referring to that will drop out the constructs.

File: src/main/java/org/alliancegenome/curation_api/services/associations/alleleAssociations/AlleleGeneAssociationService.java
Comment (oblodgett) [inline src/main/java/org/alliancegenome/curation_api/services/associations/alleleAssociations/AlleleGeneAssociationService.java]: It doesn't look like the gene or allele services are getting used?

File: src/main/java/org/alliancegenome/curation_api/jobs/util/SlackNotifier.java
Comment (markquintontulloch) [reply src/main/java/org/alliancegenome/curation_api/jobs/util/SlackNotifier.java:25]: it was put in to avoid endlessly looping errors when developing locally - it's to avoid entering the block of code that starts with `if(!systemName.get().equals("\"\"") && !slackToken.get().equals("\"\"")) {`
Does adding `NET=dev` do that?

File: src/main/java/org/alliancegenome/curation_api/model/entities/GenomicEntity.java
Comment (markquintontulloch) [reply src/main/java/org/alliancegenome/curation_api/model/entities/GenomicEntity.java:33]: Would that not be a departure from the LinkML model though?  So should we just go back to manual cleanup of xrefs?

File: src/main/resources/db/migration/v0.31.0.1__linkml_v2.1.0_update.sql
Comment (oblodgett) [inline src/main/resources/db/migration/v0.31.0.1__linkml_v2.1.0_update.sql]: I think I already have this change in via an earlier commit but probably not an issue.

File: src/main/resources/db/migration/v0.31.0.1__linkml_v2.1.0_update.sql
Comment (oblodgett) [inline src/main/resources/db/migration/v0.31.0.1__linkml_v2.1.0_update.sql:971]: Wondering if these should be "pk" vs "pkey"

**Review by oblodgett — COMMENTED** [review]
How long does the migration take?

File: src/main/java/org/alliancegenome/curation_api/dao/AffectedGenomicModelDAO.java
Comment (markquintontulloch) [reply src/main/java/org/alliancegenome/curation_api/dao/AffectedGenomicModelDAO.java]: IDs only, but we already have that method (findFilteredIds) - will replace this and similar in other DAOs

**Comment by markquintontulloch** [comment]
> How long does the migration take?

I think it was around 20 mins locally

File: src/main/java/org/alliancegenome/curation_api/services/associations/alleleAssociations/AlleleGeneAssociationService.java
Comment (markquintontulloch) [reply src/main/java/org/alliancegenome/curation_api/services/associations/alleleAssociations/AlleleGeneAssociationService.java]: Good spot - will remove

File: src/main/java/org/alliancegenome/curation_api/model/entities/GenomicEntity.java
Comment (oblodgett) [reply src/main/java/org/alliancegenome/curation_api/model/entities/GenomicEntity.java:33]: It jsut that via hibernate a OneToMany would imply that there is a table called CrossReference that has a GenomeEntityFK back to this table... that is not true in that there is a join table in between the two tables which really should be considered as a ManyToMany relationship. This works, but is not the ideal hibernate way of doing it.

File: src/main/java/org/alliancegenome/curation_api/jobs/util/SlackNotifier.java
Comment (oblodgett) [reply src/main/java/org/alliancegenome/curation_api/jobs/util/SlackNotifier.java:25]: Well the NET=dev sets it to a value as I would guess maybe your file is missing it?

**Comment by oblodgett** [comment]
> > How long does the migration take?
> 
> I think it was around 20 mins locally

NICE!!!

File: src/main/java/org/alliancegenome/curation_api/dao/AffectedGenomicModelDAO.java
Comment (oblodgett) [reply src/main/java/org/alliancegenome/curation_api/dao/AffectedGenomicModelDAO.java]: Ya not a big deal just trying to use the least amount of JQL as possible.

File: src/main/resources/db/migration/v0.31.0.1__linkml_v2.1.0_update.sql
Comment (markquintontulloch) [reply src/main/resources/db/migration/v0.31.0.1__linkml_v2.1.0_update.sql:971]: "pkey" is the default suffix that's used on all our primary keys where we don't specify a name in the table definition

File: src/main/resources/db/migration/v0.31.0.1__linkml_v2.1.0_update.sql
Comment (markquintontulloch) [reply src/main/resources/db/migration/v0.31.0.1__linkml_v2.1.0_update.sql]: yeah, will remove

**Comment by markquintontulloch** [comment]
That may be optimistic remembering, but it certainly wasn't hours!

File: src/main/resources/db/migration/v0.31.0.1__linkml_v2.1.0_update.sql
Comment (oblodgett) [reply src/main/resources/db/migration/v0.31.0.1__linkml_v2.1.0_update.sql:971]: Oh ok... I thought it was "pk" as the others are "uk" and "fk" but whatever.

**Review by oblodgett — APPROVED** [review]

**Pull request merged**
